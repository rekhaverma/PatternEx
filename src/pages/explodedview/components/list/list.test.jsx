import React from 'react';
import List from './list.component';
import moment from 'moment';

import {
  DETAILS_CARD,
  TOP3ANALYTICS_CARD,
  RELATED_SOURCE_IPS,
  RELATED_USERS,
  ANALYTICS,
  LABEL_HISTORY,
  DOMAIN_INFO,
  DOMAIN_INFO_SUBDOMAINS,
  DOMAIN_INFO_RESOLUTIONS,
  NX_DOMAINS,
  VULNERABILITY_REPORT,
} from './constants.jsx';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<List />', () => {
  it('should match with snapshot - no-data', () => {
    const props = {
      data: null,
      customClass: {},
      pipeline: '',
      columnFormat: [],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-01').endOf('day'),
      behaviorType: '',
      type: '',
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - DETAILS_CARD', () => {
    const props = {
      data: {
        predicted_prob_sum: 1,
        predicted_tag: 'test',
        ts: '2018-01-01',
      },
      customClass: {},
      pipeline: 'sip',
      columnFormat: [],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-01').endOf('day'),
      behaviorType: 'malicious',
      type: DETAILS_CARD,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - TOP3ANALYTICS_CARD', () => {
    const props = {
      data: {
        predicted_prob_sum: 1,
        predicted_tag: 'test',
        ts: '2018-01-01',
        top_n_features: [
          'demo',
        ],
      },
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-01').endOf('day'),
      behaviorType: 'malicious',
      type: TOP3ANALYTICS_CARD,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - DOMAIN_INFO', () => {
    const props = {
      data: {
        predicted_prob_sum: 1,
        predicted_tag: 'test',
        ts: '2018-01-01',
        top_n_features: [
          'demo',
        ],
      },
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-01').endOf('day'),
      behaviorType: 'malicious',
      type: DOMAIN_INFO,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - NX_DOMAINS', () => {
    const props = {
      data: [
        {
          demo: 'demo',
        }],
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-01').endOf('day'),
      behaviorType: 'malicious',
      type: NX_DOMAINS,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - DOMAIN_INFO_RESOLUTIONS', () => {
    const props = {
      data: [
        {
          demo: 'demo',
        }],
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-01').endOf('day'),
      behaviorType: 'malicious',
      type: DOMAIN_INFO_RESOLUTIONS,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - LABEL_HISTORY', () => {
    const props = {
      data: [
        {
          predicted_prob_sum: 1,
          predicted_tag: 'test',
          ts: '2018-01-01',
          top_n_features: [
            'demo',
          ],
          create_time: '2018-01-02',
        }],
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-03').endOf('day'),
      behaviorType: 'malicious',
      type: LABEL_HISTORY,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - DOMAIN_INFO_SUBDOMAINS', () => {
    const props = {
      data: [
        {
          predicted_prob_sum: 1,
          predicted_tag: 'test',
          ts: '2018-01-01',
          top_n_features: [
            'demo',
          ],
          create_time: '2018-01-02',
        }],
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-03').endOf('day'),
      behaviorType: 'malicious',
      type: DOMAIN_INFO_SUBDOMAINS,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - RELATED_SOURCE_IPS', () => {
    const props = {
      data: [
        {
          predicted_prob_sum: 1,
          predicted_tag: 'test',
          ts: '2018-01-01',
          top_n_features: [
            'demo',
          ],
          create_time: '2018-01-02',
        }],
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-03').endOf('day'),
      behaviorType: 'malicious',
      type: RELATED_SOURCE_IPS,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - RELATED_USERS', () => {
    const props = {
      data: [
        {
          predicted_prob_sum: 1,
          predicted_tag: 'test',
          ts: '2018-01-01',
          top_n_features: [
            'demo',
          ],
          create_time: '2018-01-02',
        }],
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: [],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-03').endOf('day'),
      behaviorType: 'malicious',
      type: RELATED_USERS,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - ANALYTICS', () => {
    const props = {
      data: {
        predicted_prob_sum: 1,
        predicted_tag: 'test',
        ts: '2018-01-01',
        top_n_features: [
          'demo',
        ],
        create_time: '2018-01-02',
      },
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: ['predicted_prob_sum'],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-03').endOf('day'),
      behaviorType: 'malicious',
      type: ANALYTICS,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match with snapshot - VULNERABILITY_REPORT', () => {
    const props = {
      data: {
        predicted_prob_sum: 1,
        predicted_tag: 'test',
        ts: '2018-01-01',
        top_n_features: [
          'demo',
        ],
        create_time: '2018-01-02',
      },
      customClass: {},
      pipeline: 'sip',
      columnFormat: [
        {
          label: 'demo',
        }],
      tabData: ['predicted_prob_sum'],
      handlers: {},
      startDate: moment.utc('2018-01-01').startOf('day'),
      endDate: moment.utc('2018-01-03').endOf('day'),
      behaviorType: 'malicious',
      type: VULNERABILITY_REPORT,
    };
    const wrapper = shallow(<List {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
