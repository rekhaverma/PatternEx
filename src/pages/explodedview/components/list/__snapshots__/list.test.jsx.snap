// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should match with snapshot - ANALYTICS 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "predicted_prob_sum",
        "name": "predicted_prob_sum",
        "value": "1.00",
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/analytics"
  />
</div>
`;

exports[`<List /> should match with snapshot - DETAILS_CARD 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "sip",
        "name": "Source IP",
        "value": undefined,
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/details_card"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "behavior_type",
        "name": "Alert type",
        "value": "Malicious",
      }
    }
    handlers={Object {}}
    key="listitem-1"
    type="@@listComponent/details_card"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "predicted_tag.name",
        "name": "Alert category",
        "value": <FormattedMessage
          id="evp.noDataLabel"
          values={Object {}}
        />,
      }
    }
    handlers={Object {}}
    key="listitem-2"
    type="@@listComponent/details_card"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "predicted_prob_sum",
        "name": "Attack probability",
        "value": "1.00",
      }
    }
    handlers={Object {}}
    key="listitem-3"
    type="@@listComponent/details_card"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "score",
        "name": "Outlier score",
        "value": 0,
      }
    }
    handlers={Object {}}
    key="listitem-4"
    type="@@listComponent/details_card"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "ts",
        "name": "Timestamp",
        "value": "01-01-2018 12:00 AM",
      }
    }
    handlers={Object {}}
    key="listitem-5"
    type="@@listComponent/details_card"
  />
</div>
`;

exports[`<List /> should match with snapshot - DOMAIN_INFO 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "",
        "name": "CDN Information",
        "value": <FormattedMessage
          id="evp.noDataLabel"
          values={Object {}}
        />,
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/domain_info"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "comunity_tag",
        "name": "Comunity Tag",
        "value": <FormattedMessage
          id="evp.noDataLabel"
          values={Object {}}
        />,
      }
    }
    handlers={Object {}}
    key="listitem-1"
    type="@@listComponent/domain_info"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "domain_resolution",
        "name": "Domain Resolution",
        "value": <FormattedMessage
          id="evp.noDataLabel"
          values={Object {}}
        />,
      }
    }
    handlers={Object {}}
    key="listitem-2"
    type="@@listComponent/domain_info"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "subdomains",
        "name": "Subdomains",
        "value": <FormattedMessage
          id="evp.noDataLabel"
          values={Object {}}
        />,
      }
    }
    handlers={Object {}}
    key="listitem-3"
    type="@@listComponent/domain_info"
  />
</div>
`;

exports[`<List /> should match with snapshot - DOMAIN_INFO_RESOLUTIONS 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": undefined,
        "name": undefined,
        "value": undefined,
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/domain_info_resolution"
  />
</div>
`;

exports[`<List /> should match with snapshot - DOMAIN_INFO_SUBDOMAINS 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": Object {
          "create_time": "2018-01-02",
          "predicted_prob_sum": 1,
          "predicted_tag": "test",
          "top_n_features": Array [
            "demo",
          ],
          "ts": "2018-01-01",
        },
        "name": Object {
          "create_time": "2018-01-02",
          "predicted_prob_sum": 1,
          "predicted_tag": "test",
          "top_n_features": Array [
            "demo",
          ],
          "ts": "2018-01-01",
        },
        "value": "",
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/domain_info_subdomains"
  />
</div>
`;

exports[`<List /> should match with snapshot - LABEL_HISTORY 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "canBeDeleted": true,
        "labelId": undefined,
        "name": undefined,
        "value": "01-02-2018",
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/label_history"
  />
</div>
`;

exports[`<List /> should match with snapshot - NX_DOMAINS 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": Object {
          "demo": "demo",
        },
        "name": Object {
          "demo": "demo",
        },
        "value": "",
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/nxdomains"
  />
</div>
`;

exports[`<List /> should match with snapshot - RELATED_SOURCE_IPS 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": undefined,
        "name": undefined,
        "value": "",
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/relatedSourceIPs_card"
  />
</div>
`;

exports[`<List /> should match with snapshot - RELATED_USERS 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "0",
        "name": "0",
        "value": "",
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/relatedUsers_card"
  />
</div>
`;

exports[`<List /> should match with snapshot - TOP3ANALYTICS_CARD 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "demo",
        "name": "demo",
        "value": "",
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/top3Analytics_card"
  />
</div>
`;

exports[`<List /> should match with snapshot - VULNERABILITY_REPORT 1`] = `
<div
  className="card undefined "
>
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "entityTyoe",
        "name": "Entity type",
        "value": undefined,
      }
    }
    handlers={Object {}}
    key="listitem-0"
    type="@@listComponent/vulnerability_report_card"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "dnsName",
        "name": "DNS Name",
        "value": undefined,
      }
    }
    handlers={Object {}}
    key="listitem-1"
    type="@@listComponent/vulnerability_report_card"
  />
  <ListItem
    customClass="card__item"
    data={
      Object {
        "key": "netbios",
        "name": "Netbios Name",
        "value": undefined,
      }
    }
    handlers={Object {}}
    key="listitem-2"
    type="@@listComponent/vulnerability_report_card"
  />
</div>
`;

exports[`<List /> should match with snapshot - no-data 1`] = `
<NoData
  className="nodata"
  intlDefault="No data to display"
  intlId="global.nodata"
  intlValues={Object {}}
  message=""
  withIcon={true}
/>
`;
